.github/workflows/check-base-images.yml

```
@@ -67,51 +66,30 @@ jobs:
-          fi
-          echo "Checking EE ${{ matrix.version }} image"
-          if base_image_updated hazelcast/hazelcast-enterprise:${{ matrix.version }} v${{ matrix.version }}/hazelcast-enterprise/Dockerfile; then 
-            echo "EE_NEEDS_REBUILD=yes" >> $GITHUB_ENV
-            log_info "Image EE ${{ matrix.version }} needs rebuild"
-          else
-            log_info "Image EE ${{ matrix.version }} is up-to-date"
-          fi
-          echo "Checking system package upgrades for EE ${{ matrix.version }} image"
-          if packages_updated_ee hazelcast/hazelcast-enterprise:${{ matrix.version }}; then
-            echo "EE_NEEDS_REBUILD=yes" >> $GITHUB_ENV
-            log_info "System package upgrades for EE ${{ matrix.version }} image available"
-          else
-            log_info "System packages for EE ${{ matrix.version }} image are up-to-date"
-          fi
-          echo "Checking EE NLC ${{ matrix.version }} image"
-          if base_image_updated ${NLC_IMAGE_NAME}:${{ matrix.version }} v${{ matrix.version }}/hazelcast-enterprise/Dockerfile; then 
-            echo "EE_NLC_NEEDS_REBUILD=yes" >> $GITHUB_ENV
-            log_info "Image EE NLC ${{ matrix.version }} needs rebuild"
-          else
-            log_info "Image EE NLC ${{ matrix.version }} is up-to-date"
-          fi
-          echo "Checking system package upgrades for EE NLC ${{ matrix.version }} image"
-          if packages_updated_ee ${NLC_IMAGE_NAME}:${{ matrix.version }}; then
-            echo "EE_NLC_NEEDS_REBUILD=yes" >> $GITHUB_ENV
-            log_info "System package upgrades for EE NLC ${{ matrix.version }} image available"
-          else
-            log_info "System packages for EE NLC ${{ matrix.version }} image are up-to-date"
-          fi
+          . .github/scripts/base-image-updated.functions.sh
+          . .github/scripts/packages-updated.functions.sh
+          . .github/scripts/logging.functions.sh
+
+          check_image() {
+            local label=$1
+            local image=$2
+            local dockerfile=$3
+            local env_var=$4
+
+            echo "Checking ${label} image"
+            if base_image_updated_from_dockerfile "${image}" "${dockerfile}"; then 
+              echo "${env_var}=yes" >> $GITHUB_ENV
+              echonotice "Image ${label} needs rebuild"
+            elif packages_updated_ee "${image}"; then
+              echo "${env_var}=yes" >> $GITHUB_ENV
+              echonotice "System package upgrades for ${label} image available"
+            else
+              echodebug "Image ${label} is up-to-date"
+            fi
+          }
+
+          check_image "EE" "hazelcast/hazelcast-enterprise:${{ matrix.version }}-slim" "v${{ matrix.version }}/hazelcast-enterprise/Dockerfile" "EE_NEEDS_REBUILD"
+          check_image "EE NLC" "${{ secrets.NLC_IMAGE_NAME }}:${{ matrix.version }}" "v${{ matrix.version }}/hazelcast-enterprise/Dockerfile" "EE_NLC_NEEDS_REBUILD"
```

We could drop the last parameter of `check_image` function, and reuse the first one:
```suggestion
          check_image "EE_NLC" "${{ secrets.NLC_IMAGE_NAME }}:${{ matrix.version }}" "v${{ matrix.version }}/hazelcast-enterprise/Dockerfile"
```
and in `check_image`
```shell
echo "${env_var}_NEEDS_REBUILD=yes" >> $GITHUB_ENV
```

---

https://github.com/hazelcast/hazelcast-docker/pull/1021/commits/17ff22650b1c73c735902bf70a36b806072b84fd